### PR TITLE
Login functionality

### PR DESCRIPTION
**Pull request by cosmas28**
## What does this PR do?
Implemented login functionality.

### Approach
I used **Test Driven Development** and Incremental programming throughout the challenge. You're more productive while coding, and TDD helps keep that productivity high by narrowing your focus. You'll write one failing test, and focus solely on that to get it passing. It forces you to think about smaller chunks of functionality at a time rather than the application as a whole, and you can then incrementally build on a passing test, rather than trying to tackle the bigger picture from the get-go, which will probably result in more bugs, and therefore a longer development time.

### Tasks achieved in this PR
- [x] Design snapshot tests to verify that the components actually renders
- [x] Design test specs to verify callbacks and props
- [x] Implement SignInPage presentation component
- [x] Add Routes and Link to the login page

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -1,16 +1,65 @@
...
+    handleLoginSubmit = event => {
+        event.preventDefault();
+        const input = {
+            'email': this.state.email,
+            'password': this.state.password
+        };
+        axios.post(this.apiUrl, input).then(response => {
+            this.setState({
+                'loginSuccessMessage': response.data.response_message,
+                'accessToken': response.data.access_token,
```
Comment (natalie-elizabeth) [inline src/components/App.js:37]: object keys don't have quotation  marks

File: src/components/App.js [hunk 2]
```diff
@@ -1,16 +1,65 @@
...
+        const target = event.target;
+        const value = target.value;
+        const name = target.name;
+        this.setState({ [name]: value });
+    };
+
+    handleLoginSubmit = event => {
+        event.preventDefault();
+        const input = {
+            'email': this.state.email,
```
Comment (natalie-elizabeth) [inline src/components/App.js:31]: same here

File: src/components/App.js [hunk 3]
```diff
@@ -1,16 +1,65 @@
...
+            'password': this.state.password
+        };
+        axios.post(this.apiUrl, input).then(response => {
+            this.setState({
+                'loginSuccessMessage': response.data.response_message,
```
Comment (natalie-elizabeth) [inline src/components/App.js:36]: same here

**Review by natalie-elizabeth — CHANGES_REQUESTED** [review]
Good work though.

File: src/components/App.js (hunk 1 above)
Comment (cosmas28) [reply src/components/App.js:37]: thank you for the review. I will remove quotation marks from the objects.

**Pull request merged**